html/captive-portal/templates/wireless-profile.xml

```diff
@@ -6,55 +6,145 @@
         <key>PayloadContent</key>
         <array>
                 <dict>
+                        [% IF provisioner.eap_type != "" and provisioner.security_type != "none" %]
+                        <key>AutoJoin</key>
+                        <true/>
                         <key>EAPClientConfiguration</key>
                         <dict>
                                 <key>AcceptEAPTypes</key>
                                 <array>
-                                        <integer>25</integer>
+                                        <integer>[% provisioner.eap_type %]</integer>
+                                </array>
+                                <key>PayloadCertificateAnchorUUID</key>
+                                <array>
+                                        <string>96148a46-d87a-11e4-b9d6-1681e6b88ec1</string>
+                                        <string>c83817c4-d7c4-11e4-b9d6-1681e6b88ec1</string>
+                                </array>
+                                <key>TLSTrustedServerNames</key>
+                                <array>
+                                    <string>svr</string>
                                 </array>
-                                <key>EAPFASTProvisionPAC</key>
-                                <false/>
-                                <key>EAPFASTProvisionPACAnonymously</key>
-                                <false/>
-                                <key>EAPFASTUsePAC</key>
-                                <false/>
-                                <key>TLSAllowTrustExceptions</key>
-                                <true/>
+                                [% IF for_windows %]
                                 <key>UserName</key>
                                 <string>[% username %]</string>
-                                <key>UserPassword</key>
-                                <string></string>
+                                [% END %]
                         </dict>
+                        [% END %]
                         <key>EncryptionType</key>
-                        <string>WPA</string>
+                        <string>[% provisioner.security_type %]</string>
                         <key>HIDDEN_NETWORK</key>
```

Comment:
What if it is a hidden network. Do you start seeing my point for another template ?